medicines/doc-index-updater/src/delete_manager/mod.rs

```diff
@@ -60,23 +62,41 @@ where
 
     let error_message = error.to_string();
 
-    if let ProcessMessageError::DocumentNotFoundInIndex(id) = error {
-        tracing::info!(
-            "Document {} wasn't found during delete, removing message",
-            id
-        );
-
-        state_manager
-            .set_status(
-                removeable_message.get_message().job_id,
-                JobStatus::Error {
-                    message: error_message,
-                    code: "".to_string(),
-                },
-            )
-            .await?;
-
-        let _remove = removeable_message.remove().await?;
+    match error {
+        ProcessMessageError::DocumentNotFoundInIndex(id) => {
+            tracing::info!(
+                "Document {} wasn't found during delete, removing message",
+                id
+            );
+
+            state_manager
+                .set_status(
+                    removeable_message.get_message().job_id,
+                    JobStatus::Error {
+                        message: error_message,
+                        code: "".to_string(),
+                    },
+                )
+                .await?;
+            let _remove = removeable_message.remove().await?;
+        }
+        ProcessMessageError::FailedRestoringIndex(_, _) => {
+            tracing::error!("{}", error.to_string());
+            state_manager
+                .set_status(
+                    removeable_message.get_message().job_id,
+                    JobStatus::Error {
+                        message: error.to_string(),
+                        code: "".to_string(),
+                    },
+                )
+                .await?;
+            let _remove = removeable_message.remove().await?;
```

Comment:
Just to get this clear in my head, to get here, we've:

* found blob in index
* deleted from index
* failed to delete blob
* failed to re-add to index

Right?

If so, would it be worth writing a log message here to identify which blob caused the problem so it can be investigated/manually deleted? The blob name is already the first argument to this arm.

Reply:
Valid concern! The `tracing::error` on line 84 should include the blob ID, since when this error is written to string, the blob name is included: `Cannot restore index for blob with ID <blob_id>: <wrapped error message>`